### ISSUE TITLE
Register callbacks with JavaScript function expression

### ISSUE THREAD

**Issue by lburgazzoli** [robertkrimen/otto#319]
So I'm adding support for scripting using js to my application and I want to know if it is possible to use a javascript function expression so assuming I have something like in go:

```go
type MyData struct {
}

type MyStruct struct {
}

func (target *MyStruct) Process(func(myData *DataMyData {
   ...
}
```

Then I want to use such method in my scripting, something like:

```js
mystruct.Process(data => doSomthing(data));
``` 

Is that possible ?

**Comment by dolohow** [comment]
Yes, it should work, however I only did this without passing `*DataMyData`.

Also, you need to change fat arrow to old-style `function () {}`, because ES6 is not supported.  Therefore:
```js
mystruct.Process(function (data) {
    return doSomthing(data);
);
```
